src/components/Card/Card.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import * as React from "react";
+
+import cn from "classnames";
+import CardHeader from "./CardHeader";
+import CardTitle from "./CardTitle";
+import CardBody from "./CardBody";
+
+type Props = {|
+  +children?: React.Node,
+  +className?: string,
+  +title?: string,
+  +body?: React.Node,
+  +as: React.ElementType
+|};
+
+const Card = ({
```

Comment:
Sure I switch them all up and do it this way going forward

Reply:
Actually, won't we loose the advantage of having an obvious function API/signature if we do the props destructuring in the function body?

Reply:
How so?  Not sure I understand the question.  I think this enables us to have a clean declaration and then you can do something like `const cardHeader = props.title && ( ... );` in the method body with ease.

Reply:
Not quite sure how to explain what I mean but try this...
From the outside, `function Card(props){...}` doesn't give a developer any clues as to what can be passed into the function. The types should help but the built and distributed library strips all the types out so depending on how they are using it this might not be an option for them unless I'm mistaken?

In addition, I think we would need to be exporting our types in order for editors/IntelliSense  to work correctly? I'm see this difference in Visual Studio Code ![props](https://user-images.githubusercontent.com/660222/38781922-65ec71f0-40e4-11e8-9f90-7df73507c8c8.png) vs ![destructured](https://user-images.githubusercontent.com/660222/38781924-7a683600-40e4-11e8-88b2-7dfe6417434a.png) to me the second is far more useful?

Reply:
Ah yes, I do see what you are saying.  I will look into a way to make it give more detail than that (besides exporting flow types for people who want to use flow and our project alongside theirs).  For now, I think let's agree to disagree on this one if that's okay with you - I can write my way and you can write yours as both are essentially accomplishing the same anyways and it's more a matter of personal preference.  Does that sound okay to you?